From e5395818499bd0e14815559bae25c67f3273dcdc Mon Sep 17 00:00:00 2001
From: Wangda Tan <wangda@apache.org>
Date: Mon, 4 May 2015 15:49:19 -0700
Subject: [PATCH 1961/2848] YARN-3375. NodeHealthScriptRunner.shouldRun()
 check is performing 3 times for starting
 NodeHealthScriptRunner (Devaraj K via wangda)

(cherry picked from commit 71f4de220c74bf2c90630bd0442979d92380d304)

(cherry picked from commit 4e77ee2)

Change-Id: I5dbfe56fd0864d6dc2ac9fcaa51b6b5737580487
---
 .../apache/hadoop/util/NodeHealthScriptRunner.java |    8 --------
 .../nodemanager/NodeHealthCheckerService.java      |    4 +---
 .../yarn/server/nodemanager/NodeManager.java       |    4 +++-
 3 files changed, 4 insertions(+), 12 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/NodeHealthScriptRunner.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/NodeHealthScriptRunner.java
index 568ad5b..fc392c4 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/NodeHealthScriptRunner.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/NodeHealthScriptRunner.java
@@ -214,11 +214,6 @@ protected void serviceInit(Configuration conf) throws Exception {
    */
   @Override
   protected void serviceStart() throws Exception {
-    // if health script path is not configured don't start the thread.
-    if (!shouldRun(nodeHealthScript)) {
-      LOG.info("Not starting node health monitor");
-      return;
-    }
     nodeHealthScriptScheduler = new Timer("NodeHealthMonitor-Timer", true);
     // Start the timer task immediately and
     // then periodically at interval time.
@@ -232,9 +227,6 @@ protected void serviceStart() throws Exception {
    */
   @Override
   protected void serviceStop() {
-    if (!shouldRun(nodeHealthScript)) {
-      return;
-    }
     if (nodeHealthScriptScheduler != null) {
       nodeHealthScriptScheduler.cancel();
     }
diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/NodeHealthCheckerService.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/NodeHealthCheckerService.java
index 02b318a..c1a159a 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/NodeHealthCheckerService.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/NodeHealthCheckerService.java
@@ -21,7 +21,6 @@
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.service.CompositeService;
 import org.apache.hadoop.util.NodeHealthScriptRunner;
-import org.apache.hadoop.yarn.conf.YarnConfiguration;
 
 /**
  * The class which provides functionality of checking the health of the node and
@@ -44,8 +43,7 @@ public NodeHealthCheckerService(NodeHealthScriptRunner scriptRunner,
 
   @Override
   protected void serviceInit(Configuration conf) throws Exception {
-    if (NodeHealthScriptRunner.shouldRun(
-        conf.get(YarnConfiguration.NM_HEALTH_CHECK_SCRIPT_PATH))) {
+    if (nodeHealthScriptRunner != null) {
       addService(nodeHealthScriptRunner);
     }
     addService(dirsHandler);
diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/NodeManager.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/NodeManager.java
index 6d54abf..ac78b67 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/NodeManager.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/NodeManager.java
@@ -197,7 +197,9 @@ public static NodeHealthScriptRunner getNodeHealthScriptRunner(Configuration con
     String nodeHealthScript =
         conf.get(YarnConfiguration.NM_HEALTH_CHECK_SCRIPT_PATH);
     if(!NodeHealthScriptRunner.shouldRun(nodeHealthScript)) {
-      LOG.info("Abey khali");
+      LOG.info("Node Manager health check script is not available "
+          + "or doesn't have execute permission, so not "
+          + "starting the node health script runner.");
       return null;
     }
     long nmCheckintervalTime = conf.getLong(
-- 
1.7.9.5

